<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Revisiting Cellular Diversity</title>
    <meta charset="utf-8" />
    <meta name="author" content="Yen-Chung Chen Claude Desplan Lab, NYU" />
    <script src="libs/header-attrs-2.21/header-attrs.js"></script>
    <link rel="stylesheet" href="assets/custom.css" type="text/css" />
    <link rel="stylesheet" href="assets/custom-fonts.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

.title[
# Revisiting Cellular Diversity
]
.subtitle[
## An Odessey for actionable markers
]
.author[
### Yen-Chung Chen<br/>Claude Desplan Lab, NYU
]
.date[
### 2023-08-10 @ Night Science
]

---




# About Me

.pull-left[
## Yen-Chung Chen

- 6th-year PhD student in the Desplan lab at NYU
- Work with ðŸª°, <svg aria-hidden="true" role="img" viewBox="0 0 581 512" style="height:1em;width:1.13em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:currentColor;overflow:visible;position:relative;"><path d="M581 226.6C581 119.1 450.9 32 290.5 32S0 119.1 0 226.6C0 322.4 103.3 402 239.4 418.1V480h99.1v-61.5c24.3-2.7 47.6-7.4 69.4-13.9L448 480h112l-67.4-113.7c54.5-35.4 88.4-84.9 88.4-139.7zm-466.8 14.5c0-73.5 98.9-133 220.8-133s211.9 40.7 211.9 133c0 50.1-26.5 85-70.3 106.4-2.4-1.6-4.7-2.9-6.4-3.7-10.2-5.2-27.8-10.5-27.8-10.5s86.6-6.4 86.6-92.7-90.6-87.9-90.6-87.9h-199V361c-74.1-21.5-125.2-67.1-125.2-119.9zm225.1 38.3v-55.6c57.8 0 87.8-6.8 87.8 27.3 0 36.5-38.2 28.3-87.8 28.3zm-.9 72.5H365c10.8 0 18.9 11.7 24 19.2-16.1 1.9-33 2.8-50.6 2.9v-22.1z"/></svg>, 
<svg aria-hidden="true" role="img" viewBox="0 0 448 512" style="height:1em;width:0.88em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:currentColor;overflow:visible;position:relative;"><path d="M439.8 200.5c-7.7-30.9-22.3-54.2-53.4-54.2h-40.1v47.4c0 36.8-31.2 67.8-66.8 67.8H172.7c-29.2 0-53.4 25-53.4 54.3v101.8c0 29 25.2 46 53.4 54.3 33.8 9.9 66.3 11.7 106.8 0 26.9-7.8 53.4-23.5 53.4-54.3v-40.7H226.2v-13.6h160.2c31.1 0 42.6-21.7 53.4-54.2 11.2-33.5 10.7-65.7 0-108.6zM286.2 404c11.1 0 20.1 9.1 20.1 20.3 0 11.3-9 20.4-20.1 20.4-11 0-20.1-9.2-20.1-20.4.1-11.3 9.1-20.3 20.1-20.3zM167.8 248.1h106.8c29.7 0 53.4-24.5 53.4-54.3V91.9c0-29-24.4-50.7-53.4-55.6-35.8-5.9-74.7-5.6-106.8.1-45.2 8-53.4 24.7-53.4 55.6v40.7h106.9v13.6h-147c-31.1 0-58.3 18.7-66.8 54.2-9.8 40.7-10.2 66.1 0 108.6 7.6 31.6 25.7 54.2 56.8 54.2H101v-48.8c0-35.3 30.5-66.4 66.8-66.4zm-6.7-142.6c-11.1 0-20.1-9.1-20.1-20.3.1-11.3 9-20.4 20.1-20.4 11 0 20.1 9.2 20.1 20.4s-9 20.3-20.1 20.3z"/></svg>, and C
- Learn more:
&lt;img src="slidedeck_files/figure-html/unnamed-chunk-2-1.png" width="45%" /&gt;

]

.pull-right[
&lt;img src="img/Yen.jpg" width="75%" /&gt;
]

---
layout: true
# Genetic determinism and genomics

.left-column[]

.right-column[
  &lt;img src="img/dna-4043148_1280.png" width="853" /&gt;
]

.footnote[
  Image by [Gordon Johnson](https://pixabay.com/users/gdj-1086657/?utm_source=link-attribution&amp;utm_medium=referral&amp;utm_campaign=image&amp;utm_content=4043148) from Pixabay
]

---

- 1928 Griffith's experiment

--
- 1977: Sanger sequencing

--
- 1990: Human Genome Project Started

--
- 1995: First cDNA microarray 

--
- 2000: Short-read sequencing

--
- 2006: First RNA-Seq

--
- 2009: First manual scRNA-seq

--
- 2014: First commercial scRNA-seq

---
layout: false
# What's missing?

Sometimes, the goal just seems to be moving _really_ fast... but for good
reasons.

--
.midimg[
&lt;img src="img/moving_goal.png" width="1311" /&gt;
]

---
layout: true
# What might be _NOT_ missing?
.lgimg[
.left-column[
![](img/yao_umap.jpg)&lt;!-- --&gt;
]
]
.footnote[
[Allen Brain Atlas / Yao _et al._ (2022)](https://doi.org/10.1016/j.cell.2021.04.021)
]

---
.right-column[
How many cell types are there?
]

---

.right-column[
How many cell types are there?
## 43
]

---
layout: true
# What might be _NOT_ missing?
.left-column[
.lgimg[
&lt;img src="img/ozel_umap.png" width="1315" /&gt;
]
]
.footnote[
[Ã–zel and Simon _et al._ (2021)](https://doi.org/10.1038/s41586-020-2879-3)
]

---
.right-column[
How many cell types are there?
]

---

.right-column[
How many cell types are there?
## 248
]

---
layout: false
# All's good in scRNAseq-opia?

--
.left-column[
![](img/humans-and-flies-employ-very-similar-mechanisms-for-brain-development-and-function-cropped-780x440.x9cfab47e.webp)&lt;!-- --&gt;
]
.footnote[
Image credit: [Kings College London](https://www.kcl.ac.uk/news/humans-and-flies-employ-very-similar-mechanisms-for-brain-development-and-function)
]

--
.right-column[

- Fly optic lobe:
    80,000

- Mouse cortex:
    5,000,000
]

---
# What is happening?

- **60** times more cells with **less than 20%** diversity - How does this
makes sense?

--

.center[
## Maybe they were just under clustering?
]

--
.midimg[
![](img/yes_but_actually_no.jpeg)&lt;!-- --&gt;
]

---
# How many clusters should you have?

- Premises: Prior knowledge about abundance &amp; sufficient sample size

--
.footnote[
[Stability Testing (Jean Fan)](https://jef.works/blog/2018/02/28/stability-testing/), 
[Actionable scRNA-seq clusters (Valentine Svensson)](https://www.nxn.se/valent/2018/3/5/actionable-scrna-seq-clusters)
]

- What do _you_ want from a "cell type"?

--
  - _What are the markers for each subpopulation? How can I FACS for it?_
  
--
  - _What are the differentially expressed genes marking each of our identified
  transcriptional subpopulations?_
  
--

### One should cluster only up to the point when you have _actionable_ clusters

---
# What exactly is _actionalble_ for _you_?

- Actionable markers are those:

  1. Specifically mark the cluster of interest, and
  
--
  2. Distinguishable with your handy tools (e.g., FACS, IHC, (F)ISH...)
  
--

## Basically, only those you can do experiment to prove a point about a cluster are actionable.
  
---
# Universal scarcity of an actionable marker

- Yao _et al._ (Allen Brain Atlas) had 388 clusters, but only 43 are actionable.

--

- Ã–zel and Simon _et al._ had 248 clusters, but only 86 are actionable.

--

## Despite the 7,000+ tiling regulartory element reporters in the fly community...

---
# Two markers are better than one, right?

.footnote[
[Shekhar _et al_. (2016)](https://doi.org/10.1016/j.cell.2016.07.054)
]

.pull-left[
.midimg[&lt;img src="img/shekhar_whole_tsne.png" width="915" /&gt;]
]

--

.pull-right[
![](img/shekhar_umap.jpg)&lt;!-- --&gt;
]

--

Sometimes, you look at a cluster and notice another marker provides extra 
information. Like _Cpne9_ splits cluster #15 here.

---
# Two markers are better than one, right?

.footnote[
[Shekhar _et al_. (2016)](https://doi.org/10.1016/j.cell.2016.07.054)
]

&lt;img src="img/shekhar_stain.png" width="2264" /&gt;

--

And you examine the co-expression _in vivo_.

---
class: section-title
# How has your experience been?

---
# Using multiple markers are non-trivial

- Identification of synergistic markers is rare and serentipitous most of
the time.

--

- I personally had little nice experiences with using _top_ markers together,
but of course your milleage may vary.

--

## Why it is hard to find combinations that fit our needs?

---
layout: true
# Differential expressed genes `\(\supset\)` markers

.pull-left[
### Cluster of interest
.midimg[&lt;img src="img/tmy14.png" width="947" /&gt;]
]

---

.pull-right[
| Gene | Adj p| log2FC | 
|:---:|:---:|---:|
|CG2022|0|0.86|
|CG12163|0|1.98|
|CG3940|0|2.85|
|Invadolysin|0|1.18|

.center[
.&lt;br/&gt;
.&lt;br/&gt;
.&lt;br/&gt;
]

]
---
.pull-right[
### CG2022
.midimg[&lt;img src="img/CG2022.png" width="947" /&gt;]
]

--

### DEGs could be too sparse to use as a marker.

---

.pull-right[
### CG12163
.midimg[&lt;img src="img/CG12163.png" width="947" /&gt;]
]

--

### Fold change could be misleading (or not actionable).

---
layout: false
# Marker synergization is a beast of its own

.pull-left[
### CG12163
.midimg[&lt;img src="img/CG12163.png" width="947" /&gt;]
]

--

.pull-right[
###CG3940
.midimg[&lt;img src="img/CG3940.png" width="947" /&gt;]
]

--

### If we take the top two markers, are they going to intersect _usefully_?

---
layout: true
# Top markers might not make the best team

.pull-left[
### Cluster of interest
.midimg[&lt;img src="img/tmy14.png" width="947" /&gt;]
]

---
--

.pull-right[
### CG12163 `\(\cap\)` CG3940
.midimg[&lt;img src="img/CG12163xCG3940.png" width="916" /&gt;]
]

---
.pull-right[
### CG12163 `\(\cap\)` CG3940
.midimg[&lt;img src="img/CG12163xCG3940_anno.png" width="916" /&gt;]
]

--

### They worked better than they were individually, but some cells outside the cluster are labeled and impossible to tell _in vivo_.

---
layout: false
# Poor markers could surprise with syngergy

.pull-left[
### Tj
.midimg[&lt;img src="img/tj.png" width="916" /&gt;]
]

--

.pull-right[
### Kn
.midimg[&lt;img src="img/kn.png" width="916" /&gt;]
]
--

### Individually, Tj and Kn did not make it even to the top 10 markers ranked by adjusted p values.

---
# Poor markers could surprise with syngergy

.pull-left[
### Cluster of interest
.midimg[&lt;img src="img/tmy14.png" width="947" /&gt;]
]



.pull-right[
### Tj `\(\cap\)` Kn
.midimg[&lt;img src="img/tjxkn.png" width="916" /&gt;]
]

--

### But they cooperate and outperform the combination of _top_ markers.

---
class: section-title
# How to find synergizing markers?

---
# David's approach

.left-sm-column[
.midimg[![](https://s18798.pcdn.co/desplanlab/wp-content/uploads/sites/3672/2023/02/David-scaled.jpg)&lt;!-- --&gt;]
.center[Yu-Chieh David Chen]
[.smimg[
![](slidedeck_files/figure-html/unnamed-chunk-7-17.png)&lt;!-- --&gt;
]](https://www.ycdavidchen.com/)
]

--

.right-lg-column[
.midimg[&lt;img src="img/pnas.png" width="1384" /&gt;]
.center[Check the full paper:]
[.smimg[
![](slidedeck_files/figure-html/unnamed-chunk-8-2.png)&lt;!-- --&gt;
]](https://doi.org/10.1073/pnas.2307451120)
]

---
# Rationale

.pull-left[
- Manually determine "off-target" clusters and find markers against those.
]

--
.pull-right[
### Kn
.lgimg[&lt;img src="img/kn.png" width="916" /&gt;]
]

---
# Rationale

.pull-left[
- Manually determine "off-target" clusters and find markers against those.
- Prioritize markers with existing reagents.
]

.pull-right[
]

---
# Rationale

.pull-left[
- Manually determine "off-target" clusters and find markers against those.
- Prioritize markers with existing reagents.
- Find more than one combinations to test _in vivo_.
]

--

.pull-right[
.lgimg[&lt;img src="img/dm1.png" width="993" /&gt;]
]

.footnote[
[Chen _et al_. (2023)](https://doi.org/10.1073/pnas.2307451120")
]
---
# Implementation

.pull-left[
- Binarize cluster-level gene expression
]

--
.pull-right[
![](slidedeck_files/figure-html/unnamed-chunk-9-3.png)&lt;!-- --&gt;

]
---
# Implementation

.pull-left[
- Binarize cluster-level gene expression
- Find double-positive clusters for each combination
]

.pull-right[
.lgimg[&lt;img src="img/dbp.png" width="1204" /&gt;]
]

---
# Implementation

.pull-left[
- Binarize cluster-level gene expression
- Find double-positive clusters for each combination
- Find specific combinations for each cluster
]

--

.pull-right[
.midimg[&lt;img src="img/identifiable_clusters.png" width="567" /&gt;]
]

.footnote[
[Chen _et al_. (2023)](https://doi.org/10.1073/pnas.2307451120")
]

--

### # of actionable clusters goes from 86 -&gt; 185!

---
# Brief summary

- It is challenging to define "off-target" expression of a marker.

--

- Comprehensive examination of all marker combinations is computationally 
too demanding to use.

--

## Binarization addresses both

--

- Off-target = Being "ON" in other clusters

--

- Binarization further reduce computational load by the ability to perform
bit-wise computation and to exclude non-binary markers (optional).

---
class: section-title
# Are we ready to apply this to every existing dataset?

---
# No, unfortunately.

- It is still demanding to examine binarized marker combination.

--

- Cluster-level binarization assumes clustering is _correct_.

--

- Cluster-level binarization requires &gt; 100 clusters to stabilize.

--

### A greedy search is approximately the same as a comprehensive one (in our data).

---
# A universal way to find marker combos

--

- Can we just do binarization at single-cell level?

--

  - Imputation
  
--

  - Cluster-constrained mixture modeling

--

- Can we have a better algorithm to find marker combinations?

--

  - Regularization (e.g., LASSO)
  
--

  - Joint information estimation
  
---
# Take-home messages

--

- Diversity is underappreciated despite accumulating single-cell
transcriptomics.

--

- Be strategic when you need to find another marker because one marker does not
work. 

--

- With the advent of AAV and other vectors, manipulation with a combination of
markers is possible in mammals as well
(e.g., [Single-vector AAV intersection](https://www.biorxiv.org/content/10.1101/2023.02.07.527312v1.full)).

---
class: section-title
# Questions?
---
class: section-title
# Thank you for your attention!
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="assets/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false,
"ratio": "16:9"
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
