---
title: Using Limma to find differentially expressed genes
author: Yen-Chung Chen
date: '2018-05-19'
slug: using-limma-to-find-differentially-expressed-genes
categories:
  - R
tags:
  - NGS
  - R
image:
  caption: ''
  focal_point: ''
---
> Ritchie, ME, Phipson, B, Wu, D, Hu, Y, Law, CW, Shi, W, and Smyth, GK
> (2015). [limma powers differential expression analyses for
> RNA-sequencing and microarray
> studies](https://www.ncbi.nlm.nih.gov/pubmed/25605792).Nucleic Acids
> Research 43(7), e47.

[limma](https://bioconductor.org/packages/release/bioc/html/limma.html)
is an R package hosted on Bioconductor which finds differentially
expressed genes for RNA-seq or microarray. Recently I’ve been working on
a PCR-based low-density array and noticed that I forgot how to use limma
for the one hundredth time, so I decided to make a note.

## Preparation

  - Log-transformed expression data in a matrix:  
    Each column represents an experiement, and each row represents a
    detected gene/probe.
  - Design matrix:  
    Each column represents a status of your data (e.g., wild-type,
    mutant, rescued…), and each row corresponds to an experiment in the
    expression matrix. This could be generated from an
    `expressionDataset` object, or we could create this manually if you
    only have a raw expression dataset like me.
  - Contrast matrix: 
    Specified a set of parameters to specify the contrasts being compared later.  

```{r}
library(limma)

# Create a dummy expression matrix
exp_matrix <- data.frame(sample_1 = c(3, 3, 2, 2, 1), 
                         sample_2 = c(3.2, 2.8, 2, 2, 1.3), 
                         sample_3 = c(1, 1, 2, 2.3, 5), 
                         sample_4 = c(1.1, 0.9, 2, 2, 4), 
                         sample_5 = c(2.5, 2.5, 2, 2.5, 2), 
                         sample_6 = c(2.6, 2.3, 2.1, 2.2, 1.8), 
                         row.names = paste("gene", seq(5), sep = "_"), 
                         stringsAsFactors = FALSE)
print(exp_matrix)

# Create a design matrix
design <- model.matrix(~ 0+factor(c(1,1,2,2,3,3)))
colnames(design) <- c("Wild_type", "Mutant", "Rescue")

print(design)

# Create a design matrix to indicate the comparisons to make by limma
cont_matrix <- makeContrasts(MutvsWT = Mutant-Wild_type, 
                             MutvsRes = Mutant-Rescue, 
                             levels=design)

print(cont_matrix)
```



## Analysis

Now, I am pretty much ready (except for the need to understand the
math).

```{r}
# Fit the expression matrix to a linear model
fit <- lmFit(exp_matrix, design)

# Compute contrast
fit_contrast <- contrasts.fit(fit, cont_matrix)

# Bayes statistics of differential expression
# *There are several options to tweak!*
fit_contrast <- eBayes(fit_contrast)

# Generate a vocalno plot to visualize differential expression
# Highlighting the top 2 genes
volcanoplot(fit_contrast, highlight = 2)
```

``` {r}
# Generate a list of top 100 differentially expressed genes
top_genes <- topTable(fit_contrast, number = 100, adjust = "BH")
print(top_genes)

# Summary of results (number of differentially expressed genes)
result <- decideTests(fit_contrast)
summary(result)
```
