+++
title = "Diverse cell-specific patterns of alternative polyadenylation in Drosophila"
date = 2022-09-13T00:00:00

# Authors. Comma separated list, e.g. `["Bob Smith", "David Jones"]`.
authors = ["Seungjae Lee", "Yen-Chung Chen", "FCA Consortium", "Austin E. Gillen", "J. Matthew Taliaferro", "Bart Deplancke", "Hongjie Li", "Eric C. Lai"] 

# Publication type.
# Legend:
# 0 = Uncategorized
# 1 = Conference paper
# 2 = Journal article
# 3 = Manuscript
# 4 = Report
# 5 = Book
# 6 = Book section
publication_types = ["2"]

# Publication name and optional abbreviated version.
publication = "In *Nature Communications*"
publication_short = "In *Nat. Commun*"

# Abstract.
abstract = "Most genes in higher eukaryotes express isoforms with distinct 3’ untranslated regions (3’ UTRs), generated by alternative polyadenylation (APA). Since 3’ UTRs are predominant locations of post-transcriptional regulation, APA can render such programs conditional, and can also alter protein sequences via alternative last exon (ALE) isoforms. We previously used 3’-sequencing from diverse Drosophila samples to define multiple tissue-specific APA landscapes. Here, we exploit comprehensive single nucleus RNA-sequencing data (Fly Cell Atlas) to elucidate cell-type expression of 3’ UTRs across >250 adult Drosophila cell types. We reveal the cellular bases of multiple tissue-specific APA/ALE programs, such as 3’ UTR lengthening in differentiated neurons and 3’ UTR shortening in spermatocytes and spermatids. We trace dynamic 3’ UTR patterns across cell lineages, including in the male germline, and discover new APA patterns in the intestinal stem cell lineage. Finally, we correlate expression of RNA binding proteins (RBPs), miRNAs and global levels of cleavage and polyadenylation (CPA) factors in several cell types that exhibit characteristic APA landscapes, yielding candidate regulators of transcriptome complexity. These analyses provide a comprehensive foundation for future investigations of mechanisms and biological impacts of alternative 3’ isoforms across the major cell types of this widely-studied model organism."

# Summary. An optional shortened abstract.
# summary = ""

# Digital Object Identifier (DOI)
doi = "https://doi.org/10.1038/s41467-022-32305-0"

# Is this a featured publication? (true/false)
featured = true

# Tags (optional).
#   Set `tags = []` for no tags, or use the form `tags = ["A Tag", "Another Tag"]` for one or more tags.
tags = ["3' UTR", "Polyadenylation", "Single-cell genomics"]

# Projects (optional).
#   Associate this publication with one or more of your projects.
#   Simply enter your project's folder or file name without extension.
#   E.g. `projects = ["deep-learning"]` references 
#   `content/project/deep-learning/index.md`.
#   Otherwise, set `projects = []`.
projects = ["Applied genomics"]

# Links (optional).
url_pdf = ""
url_preprint = ""
url_code = ""
url_dataset = ""
url_project = ""
url_slides = ""
url_video = ""
url_poster = ""
url_source = ""

# Custom links (optional).
#   Uncomment line below to enable. For multiple links, use the form `[{...}, {...}, {...}]`.
# links = [{name = "Custom Link", url = "http://example.org"}]

# Does this page contain LaTeX math? (true/false)
math = false

# Featured image
# To use, add an image named `featured.jpg/png` to your page's folder. 
[image]
  # Caption (optional)
  caption = "Figure 2 from [Lee, S., Chen, YC., FCA Consortium. et al. (2022)](https://doi.org/10.1038/s41467-022-32305-0)"

  # Focal point (optional)
  # Options: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight
  focal_point = ""
  preview_only = true  
+++
