+++
title = "Using single-cell RNA sequencing to generate cell-type-specific split-GAL4 reagents throughout development"
date = 2023-07-31T00:00:00

# Authors. Comma separated list, e.g. `["Bob Smith", "David Jones"]`.
authors = ["Yu-Chieh David Chen", "Yen-Chung Chen", "Raghuvanshi Rajesh", "Nathalie Shoji", "Maisha Jacy", "Haluk Lacin", "Ted Erclik", "Claude Desplan"] 


# Publication type.
# Legend:
# 0 = Uncategorized
# 1 = Conference paper
# 2 = Journal article
# 3 = Manuscript
# 4 = Report
# 5 = Book
# 6 = Book section
publication_types = ["2"]

# Publication name and optional abbreviated version.
publication = "Proceedings of the National Academy of Sciences"
publication_short = "In *PNAS*"

# Abstract.
abstract = "Cell-type-specific tools facilitate the identification and functional characterization of the distinct cell types that form the complexity of neuronal circuits. A large collection of existing genetic tools in Drosophila relies on enhancer activity to label different subsets of cells and has been extremely useful in analyzing functional circuits in adults. However, these enhancer-based GAL4 lines often do not reflect the expression of nearby gene(s) as they only represent a small portion of the full gene regulatory elements. While genetic intersectional techniques such as the split-GAL4 system further improve cell-type-specificity, it requires significant time and resources to screen through combinations of enhancer expression patterns. Here, we use existing developmental single-cell RNA sequencing (scRNAseq) datasets to select gene pairs for split-GAL4 and provide a highly efficient and predictive pipeline (scMarco) to generate cell-type-specific split-GAL4 lines at any time during development, based on the native gene regulatory elements. These gene-specific split-GAL4 lines can be generated from a large collection of coding intronic MiMIC/CRIMIC lines or by CRISPR knock-in. We use the developing Drosophila visual system as a model to demonstrate the high predictive power of scRNAseq-guided gene-specific split-GAL4 lines in targeting known cell types, annotating clusters in scRNAseq datasets as well as in identifying novel cell types. Lastly, the gene-specific split-GAL4 lines are broadly applicable to any other Drosophila tissue. Our work opens new avenues for generating cell-type-specific tools for the targeted manipulation of distinct cell types throughout development and represents a valuable resource for the Drosophila community."

# Summary. An optional shortened abstract.
# summary = ""

# Digital Object Identifier (DOI)
doi = "https://doi.org/10.1073/pnas.2307451120"

# Is this a featured publication? (true/false)
featured = true

# Tags (optional).
#   Set `tags = []` for no tags, or use the form `tags = ["A Tag", "Another Tag"]` for one or more tags.
tags = ["Neural development", "Drosophila melanogaster", "Marker discovery", "Split-Gal4"]

# Projects (optional).
#   Associate this publication with one or more of your projects.
#   Simply enter your project's folder or file name without extension.
#   E.g. `projects = ["deep-learning"]` references 
#   `content/project/deep-learning/index.md`.
#   Otherwise, set `projects = []`.
projects = ["Spatiotemporal patterning in the visual system"]

# Links (optional).
url_pdf = ""
url_preprint = ""
url_code = ""
url_dataset = ""
url_project = ""
url_slides = ""
url_video = ""
url_poster = ""
url_source = ""

# Custom links (optional).
#   Uncomment line below to enable. For multiple links, use the form `[{...}, {...}, {...}]`.
# links = [{name = "Custom Link", url = "http://example.org"}]

# Does this page contain LaTeX math? (true/false)
math = false

# Featured image
# To use, add an image named `featured.jpg/png` to your page's folder. 
[image]
  # Caption (optional)
  caption = "From figure 1 of Chen et al."

  # Focal point (optional)
  # Options: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight
  focal_point = ""
  preview_only = true
+++
