+++
title = "Using single-cell RNA sequencing to generate cell-type-specific split-GAL4 reagents throughout development"
date = 2023-02-04T00:00:00

# Authors. Comma separated list, e.g. `["Bob Smith", "David Jones"]`.
authors = ["Yu-Chieh David Chen", "Yen-Chung Chen", "Raghuvanshi Rajesh", "Nathalie Shoji", "Maisha Jacy", "Haluk Lacin", "Ted Erclik", "Claude Desplan"] 


# Publication type.
# Legend:
# 0 = Uncategorized
# 1 = Conference paper
# 2 = Journal article
# 3 = Manuscript
# 4 = Report
# 5 = Book
# 6 = Book section
publication_types = ["3"]

# Publication name and optional abbreviated version.
# publication = "Preprint in *bioRxiv*"
# publication_short = "Preprint in *bioRxiv*"

# Abstract.
abstract = "Cell-type-specific tools facilitate the identification and functional characterization of distinct cell types, which underly the complexity of neuronal circuits. A large collection of existing genetic tools in Drosophila relies on enhancer activity to label different subsets of cells. These enhancer-based GAL4 lines often fail to show a predicable expression pattern to reflect the expression of nearby gene(s), partly due to an incomplete capture of the full gene regulatory elements. While genetic intersectional technique such as the split-GAL4 system further improve cell-type-specificity, it requires significant time and resource to generate and screen through combinations of enhancer expression patterns. In addition, since existing enhancer-based split-GAL4 lines that show cell-type-specific labeling in adult are not necessarily active nor specific in early development, there is a relative lack of tools for the study of neural development. Here, we use an existing single-cell RNA sequencing (scRNAseq) dataset to select gene pairs and provide an efficient pipeline to generate cell-type-specific split-GAL4 lines based on the native genetic regulatory elements. These gene-specific split-GAL4 lines can be generated from a large collection of coding intronic MiMIC/CRIMIC lines either by embryo injection or in vivo cassette swapping crosses and/or CRISPR knock-in at the N or C terminal of the gene. We use the developing Drosophila visual system as a model to demonstrate the high prediction power of scRNAseq-guided gene specific split-GAL4 lines in targeting known cell types. The toolkit allows efficient cluster annotation in scRNAseq datasets but also the identification of novel cell types. Lastly, the gene-specific split-GAL4 lines are broadly applicable to Drosophila tissues. Our work opens new avenues for generating cell-type-specific tools for the targeted manipulation of distinct cell types throughout development and represents a valuable resource to the fly research community."

# Summary. An optional shortened abstract.
# summary = ""

# Digital Object Identifier (DOI)
doi = "https://doi.org/10.1101/2023.02.03.527019"

# Is this a featured publication? (true/false)
featured = true

# Tags (optional).
#   Set `tags = []` for no tags, or use the form `tags = ["A Tag", "Another Tag"]` for one or more tags.
tags = ["Neural development", "Drosophila melanogaster", "Marker discovery", "Split-Gal4"]

# Projects (optional).
#   Associate this publication with one or more of your projects.
#   Simply enter your project's folder or file name without extension.
#   E.g. `projects = ["deep-learning"]` references 
#   `content/project/deep-learning/index.md`.
#   Otherwise, set `projects = []`.
projects = ["Spatiotemporal patterning in the visual system"]

# Links (optional).
url_pdf = ""
url_preprint = ""
url_code = ""
url_dataset = ""
url_project = ""
url_slides = ""
url_video = ""
url_poster = ""
url_source = ""

# Custom links (optional).
#   Uncomment line below to enable. For multiple links, use the form `[{...}, {...}, {...}]`.
# links = [{name = "Custom Link", url = "http://example.org"}]

# Does this page contain LaTeX math? (true/false)
math = false

# Featured image
# To use, add an image named `featured.jpg/png` to your page's folder. 
[image]
  # Caption (optional)
  caption = "From figure 1 of Chen et al."

  # Focal point (optional)
  # Options: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight
  focal_point = ""
  preview_only = true
+++
